### PR TITLE
feat: style sorting icons

### PR DESCRIPTION
**Pull request by defuncart**
Add ability to style sorting icons.

### CODE REVIEW

File: lib/data_grid.dart [hunk 1]
```diff
@@ -629,3 +642,47 @@ enum SortingState {
...
+
+  const SortingIconSettings({
+    required this.ascending,
+    required this.descending,
+    required this.size,
+    this.color,
+    required this.padding,
+  });
+
+  const SortingIconSettings.defaults()
```
Comment (ted-marozzi) [inline lib/data_grid.dart:661]: I think we just give the normal constructor default values instead of two constructors

File: lib/data_grid.dart [hunk 2]
```diff
@@ -629,3 +642,47 @@ enum SortingState {
...
+        other.ascending == ascending &&
+        other.descending == descending &&
+        other.size == size &&
+        other.color == color &&
+        other.padding == padding;
+  }
+
+  @override
+  int get hashCode {
+    return ascending.hashCode ^
```
Comment (ted-marozzi) [inline lib/data_grid.dart:682]: https://api.dart.dev/stable/2.18.4/dart-core/Object/hash.html

Object.hash(...) provides a better hash I believe. I wonder if we really need to override these operators. I don't see them being compared anywhere

**Review by ted-marozzi — CHANGES_REQUESTED** [review]
Thanks for the contribution once again! Just a few comments. Much appreciated.

File: lib/data_grid.dart (hunk 2 above)
Comment (defuncart) [reply lib/data_grid.dart:682]: It is to reduce build calls. In the case of `SortingIconSettings` not changing between builds, the hash code would be a different memory address and thus an unneeded rebuild.

File: lib/data_grid.dart (hunk 1 above)
Comment (defuncart) [reply lib/data_grid.dart:661]: Ok will do :)

File: lib/data_grid.dart (hunk 2 above)
Comment (ted-marozzi) [reply lib/data_grid.dart:675]: Huh never thought of that. Good point.

File: lib/data_grid.dart [hunk 3]
```diff
@@ -629,3 +642,40 @@ enum SortingState {
...
+        descending.hashCode ^
+        size.hashCode ^
+        color.hashCode ^
+        padding.hashCode;
```
Comment (ted-marozzi) [inline lib/data_grid.dart:679]:
```suggestion
    return Object.hash(ascending, descending, size, color, padding);
```
I think this is better, the package freezed uses it internally. Syntax might be wrong as I just did this in github

**Review by ted-marozzi — CHANGES_REQUESTED** [review]
I think we need to update this func as well: https://github.com/ted-marozzi/data-grid/blob/main/lib/data_grid.dart#L210-L211

I should put some unit tests around it too if it doesn't have them already, its fine for you to omit the tests if they are missing.

**Comment by defuncart** [comment]
> I think we need to update this func as well: https://github.com/ted-marozzi/data-grid/blob/main/lib/data_grid.dart#L210-L211

Good catch, updated. Let me know if there's anything else to do.

**Review by ted-marozzi — APPROVED** [review]
Lgtm 🚀 I’m interested, are you building something in particular with this? I want to create a small roadmap for data-grid before before 1.0.0 and it might help to know what your using it for.

Things I am thinking of doing:
* Document everything properly 
* Refactor code structure, particularly the file split
* See if I can implement with slivers instead of list views. (Never used slivers so would need to up skill here)
* Add more tests
Let me know if you have other ideas

**Comment by defuncart** [comment]
> 

For a client, I am currently using this widget to display a simple list of data which is sortable. I would suggest to add an issue 'Roadmap' in which all discussions could be housed in one place :)

**Pull request merged**